### PR TITLE
fixup! Use fast decimal division for divide and modulo operators

### PR DESCRIPTION
**Pull request by arhimondr**

### CODE REVIEW

**Comment by pnowojski** [comment]
lgtm

File: presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java [hunk 1]
```diff
@@ -563,7 +563,9 @@ public static int compare(long leftRawLo, long leftRawHi, long rightRawLo, long
             return leftStrictlyNegative ? -1 : 1;
         }
         else {
-            return compareUnsigned(leftRawLo, leftRawHi, rightRawLo, rightRawHi) * (leftStrictlyNegative ? -1 : 1);
+            long leftHi = unpackUnsignedLong(leftRawHi);
+            long rightHi = unpackUnsignedLong(rightRawHi);
+            return compareUnsigned(leftRawLo, leftHi, rightRawLo, rightHi) * (leftStrictlyNegative ? -1 : 1);
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java:568]: shouldn't `compareUnsigned` handle raw low/high itself?

File: presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java [hunk 2]
```diff
@@ -586,16 +588,12 @@ public static int compareUnsigned(Slice left, Slice right)
 
     public static int compareUnsigned(long leftRawLo, long leftRawHi, long rightRawLo, long rightRawHi)
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java:589]: `leftRawHi`, `rightRawHi` is no longer raw.
raw means it has sign bit.

**Comment by sopel39** [comment]
lgtm % comments

File: presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java (hunk 1 above)
Comment (ilfrin) [reply presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java:568]: +1

File: presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java (hunk 2 above)
Comment (ilfrin) [reply presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java:589]: I'd rather leave all arguments to be in the same form and unpack internally in the method, as @sopel39 suggested in another comment

Comment (arhimondr) [reply presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java:589]: It actually does. It is only treated as a part of a number. This method works exactly as `Long.compareUnsigned`

File: presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java (hunk 1 above)
Comment (arhimondr) [reply presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java:568]: It is more intuitive when `compareUnsigned` method behaves in the same way as `Long.compareUnsigned` does. I would rather rename the `Slice` version to `compareIgnoreSign` or `compareAbsoluteValue`. From my perspective it is better reflects the purpose of this method.

**Pull request merged**